### ISSUE TITLE
Allow different font for export (PDF export crops off the tops of lines)

### ISSUE THREAD

**Issue by johnblommers** [vkbo/novelWriter#230]
PDF export crops off the tops of the lines both in the headings and every line of text. This is a screen shot showing the problem at 200% magnification when viewing the PDF with Document Viewer. This cropping does **not **occur when viewing the PDF with Foxit Reader or with Okular. I'm not sure where  the accusing finger should be pointing!

![Trimming](https://user-images.githubusercontent.com/8484427/82781568-fa0b5980-9e0e-11ea-9c0c-591f747eb126.png)

**Comment by vkbo** [comment]
The PDF export is simply a print-to-pdf that bypasses the print dialog. This is handled entirely by the Qt libraries, and seems to me to be an issue with framing the font in the text boxes.

It is not something I can fix unfortunately, unless there are some settings to tweak, but there aren't that many. It may even be an issue with the font you use. I could perhaps allow for a different font for exports than for the editor though. Currently it copies over the font settings.

**Comment by johnblommers** [comment]
As you suggested, changing the font to simply *Ubuntu* erased the problem. To reproduce this issue just use *Ubuntu Mono* as the default font. As you say, this is not something novelWriter can fix. The issue/problem lies elsewhere.

Using that Ubuntu font, damn, but that PDF output looks mighty fine!

**Comment by vkbo** [comment]
Yeah, I'm pretty sure this is a font issue, maybe font+QPainter issue.

Qt's QTextDocument does a decent print, and that is what the PDF is made from. I had to battle a bit with it to get it to apply the page breaks I wanted, but it should be working now,

The Qt print feature is a major reason why I switched to this method of export rather than the Pandoc path. The Pandoc path is still available via HTML or Markown export, just not integrated. The HTML export is still the most complete export format though as the conversion is done by novelWriter itself, and therefore has a lot more features supported.

Unfortunately, the Qt writer for Open Documents is pretty half-assed I'd say. The result is similar to the preview, but it doesn't actually export the headings as headings! I have no idea why. I may write my own `.odt` export tool at some point, or maybe just `.fodt` which is a flat XML file, but for now this will have to do. You get a pretty good result if you just export to HTML and import the file in Libre Office instead.

Being able to set a different export font is still useful, so I'm leaving the issue open and changing the title.
